docs/DeveloperGuide.md

```diff
@@ -4,26 +4,207 @@
 
-## Design & implementation
+# 1. Overview
+MeetingJio is a program for **finding potential time slots for team meetings based on everyone’s availability**.
+
+# 2. Design 
+
+## 2.1 Architecture
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## 2.2 UI Component
+
+## 2.3 Logic Component
+- Parsers
+- Commands
+
+## 2.4 Model Component
+- Timetables
+- Events
+
+## 2.5 Storage Component
+
+# 3. Implementation
+This section describes some noteworthy details on how certain features are implemented.
+
+## 3.1 Add Feature
+The `add` command is a command that allows user to 
+1. add new users to the master timetable.
+2. add new events to the timetables.
+
+The following sequence diagram shows how the `add` operation generally works.
+
+![AddCommandSequenceDiagram](diagrams/AddCommandSequenceDiagram.png)
+
+**_Note:_**
+
+This is a condensed diagram. Several terms in the sequence diagram above have been substituted by a common term.
+
+Common Term | User-specific term | Lesson-specific term | Meeting-specific term      |
+------------|-------------------|----------------------|----------------------------|        
+add | `"add_user ..."`  | `"add_lesson ..."`   | `"add_meeting ..."`     
+AddCommand | `AddUserCommand`  | `AddLessonCommand`   | `AddMeetingCommand`  
+
+
+## 3.1.1 Add User Feature
+`add_user` adds new user and his or her timetable into the master timetable.
+
+The following sequence diagram shows how the `add_user` command works:
+![AddUserCommandSequenceDiagram](diagrams/AddUserCommandSequenceDiagram.png)
+
+
+## 3.1.2 Add Events Feature
+
+`add_lesson` allows users to create a new lesson and add it to their own timetables.
+
+`add_meeting` allows users to create a new meeting and add it to all the existing timetables.
+
+
+The following sequence diagram shows how the `add_lesson` operation works in detail.
+![AddMeetingCommandSequenceDiagram](diagrams/AddLessonCommandSequenceDiagram.png)
+
+Given below is the steps on how the `add_lesson` operation works.
+
+>To be added
+
+The following sequence diagram shows how the `add_meeting` operation works in detail.
+![AddMeetingCommandSequence](diagrams/AddMeetingCommandSequence.png)
+
+Given below is the steps on how the `add_meeting` operation works.
+
+>To be added
+> 
+## 3.2 Listing Events Feature
+The `list` command is a command that the user can input in order to list out the events he has in his timetable.
+
+`list [user]` displays the timetable for the particular user.
+
+`list all` displays the timetable for all users.
+
+In essence, the `list all` command repeatedly calls `list [user]` for all users.
+
+Before the timetable is listed out, it will also be sorted according to day and time for easy reading.
+
+The following sequence diagram shows how the command `` is executed.
+![ListCommandSequenceDiagram](diagrams/ListCommandSequenceDiagram.png)
```

Comment:
(Not too sure)
Can you remove the return types when constructing an object? I think you can omit return types when calling constructors.
![image](https://user-images.githubusercontent.com/26570382/161184461-1c92a5a4-79fe-4c1f-91b2-5378d0aac830.png)